content/rest-api/index.textile

```diff
@@ -616,11 +614,17 @@ An unsuccessful request returns an error.
 
 h3(#update-device-registration). Update a device registration
 
+Device registrations can be either upserted (the existing registration is replaced entirely) with a @PUT@ operation, or specific attributes of an existing registration can be updated using a @PATCH@ operation:
+
 h6. PUT rest.ably.io/push/deviceRegistrations/@<device id>@
 
-The body must have the same shape as when "registering the device":#post-device-registration, except only fields to be changed should be provided. The @PUT@ operation is in fact a @PATCH@ operation i.e. only fields provided are updated.
+The body must have the same shape as when "registering the device":#post-device-registration. The @PUT@ operation will replace the existing device registration, however please bear in mind that a registered device in Ably is largely immutable. As such, only the following attributes are currently updateable and any attempt to modify other fields will result in the update failing:
+
+* @clientId@
+* @metadata@
+* @push.recipient@
 
-Only @metadata@ and @push.recipient@ can be updated. If non-null, the provided new values will be merged with their old values.
+Additionally, if the @push.state@ or @updateToken@ attributes are provided, they will be accepted if they match the existing value. However if they differ (and are not @null@ or @omitted@), then the update will fail.
```

Comment:
That isn't just for push.state or updateToken; all non-updatable (but settable) fields must match. Otherwise, this would really be a PATCH. In short, the body must be a valid POST body too, since we're inserting a new registration if it doesn't exist.

> (and are not @null@ or @omitted@)

This isn't true, @null@ is treated like a value just any other. There's (thankfully) no @undefined@. See my point above about omitted fields.

(Also, omitted in monospaced makes it look like a special value or something like that.)